main/field.c

```diff
@@ -385,29 +385,33 @@ static const char *renderEscapedName (const char* s,
 				      const tagEntryInfo *const tag,
 				      vString* b)
 {
-	const char* base = s;
+	int unexpected_byte = 0;
 
-	for (; *s; s++)
+	if (*s == ' ' || *s == '!')
 	{
-		int c = *s;
-		if ((c > 0x00 && c <= 0x1F) || c == 0x7F)
-		{
-			verbose ("Unexpected character (0 < *c && *c < 0x20) included in a tagEntryInfo: %s\n", base);
-			verbose ("File: %s, Line: %lu, Lang: %s, Kind: %c\n",
-				 tag->inputFileName, tag->lineNumber, getLanguageName(tag->langType), tag->kind->letter);
-			verbose ("Escape the character\n");
-			break;
-		}
-		else if (c == '\\')
-			break;
```

Comment:
@masatake why no warning when there is a `\`?  I wasn't emitting a warning for the scope `"\\Broken\tContext"` in the self-test as it stopped at the leading `\` even though there is a `\t`, but I hardly see the point here; should I still make my change behave the same?  I would rather think that this was an implementation detail and that emitting the warning in this case is the expected behavior, what do you think?

Reply:
```
if ((c > 0x00 && c <= 0x1F) || c == 0x7F)
		{
			verbose ("Unexpected character (0 < *c && *c < 0x20) included in a tagEntryInfo: %s\n", base);
			verbose ("File: %s, Line: %lu, Lang: %s, Kind: %c\n",
				 tag->inputFileName, tag->lineNumber, getLanguageName(tag->langType), tag->kind->letter);
			verbose ("Escape the character\n");
			break;
		}
		else if (c == '\\')
			break;
		else
			continue;
```
c in ` ((c > 0x00 && c <= 0x1F) || c == 0x7F)` is not printable, so ctags makes the warning.
Passing such c to the main part from a parser may be a bug. So ctags make the warning.

In other hand `\` is a printable. Passing `\` to the main part from a parser is valid. There for
ctags doesn't emit a warning. Actually `\` is a separator in php. However, ctags must convert `\' to `\\`.
`break` in `(c == '\\')` makes it possible to pass `s` including `\\` to renderEscapedString called at the
end of renderEscapedName. 

Do I answer your question?

Reply:
I read your code and understand how you mean "implementation detail".
You are correct. Handling the backslash here is just for optimization.